### ISSUE TITLE
Deadlock: IDE hangs after copy/paste files with "Updating indexes" dialog

### ISSUE THREAD

**Issue by CrazyCoder** [karollewandowski/aem-intellij-plugin#174]
AEM plug-in causes the deadlock in IntelliJ IDEA: https://youtrack.jetbrains.com/issue/IDEA-222814 .

Comment from responsible developer:

CopyFilesOrDirectoriesHandler wait for dumb tasks while holding write action, CacheUpdateRunner waits for FileContentQueue which invoked isMyFileType from aY class (which belongs to some third-party plugin) which invoked runReadAction and caused a dead lock. It looks not good to use runReadAction in isMyFileType implementation but I'm not sure how we can prevent such problems in the platform. Maybe we shouldn't wait for dumb tasks while holding write action?

Thread dump: https://pastebin.com/raw/iZzkPb6B.

**Comment by karollewandowski** [comment]
Thanks a lot for letting me know, I will try to fix it ASAP.

**Comment by karollewandowski** [comment]
I invoked `runReadAction()` in `FileTypeIdentifiableByVirtualFile.isMyFileType(file: VirtualFile)` because I wanted to `guessProject()` for `VirtualFile` (project contains information about "my specific files" source roots) and it was required to be wrapped in read action. I found:
https://intellij-support.jetbrains.com/hc/en-us/community/posts/206132529-How-do-I-get-current-Project-instance-having-VirtualFile-instance-only-

and now understand that it was totally wrong as `VirtualFile` is application and not project concept. I easily got rid of `isMyFileType()` method which was created for filtering file during indexing and I filter "not my files" in index client code.

The bug was introduced about 1 year ago.
I fixed it and uploaded the new version to the plugins repository.

Thanks again for that!
